projects/addon-mobile/directives/ripple/ripple.directive.ts

```diff
@@ -1,79 +1,158 @@
+        const touchEnd$ = merge(
+            tuiTypedFromEvent(element, 'touchend'),
+            tuiTypedFromEvent(element, 'mouseup'),
+            tuiTypedFromEvent(element, 'mouseout'),
+        );
+        const touchMove$ = merge(
+            tuiTypedFromEvent(element, 'touchmove'),
+            tuiTypedFromEvent(element, 'mousemove'),
+        );
+
+        race(TOUCH_MOVE_DELAY_TIMER, touchEnd$.pipe(map(TUI_TRUE_HANDLER)))
             .pipe(
-                mergeMap(ripple => {
-                    const animationEndOn$ = tuiTypedFromEvent(ripple, 'animationend');
-
-                    return race(
-                        timer(TOUCH_MOVE_DELAY).pipe(map(TUI_FALSE_HANDLER)),
-                        touchEnd$.pipe(map(TUI_TRUE_HANDLER)),
-                    ).pipe(
-                        take(1),
-                        // eslint-disable-next-line rxjs/no-unsafe-takeuntil
-                        takeUntil(touchMove$),
-                        tap(() => {
-                            this.renderer.setStyle(
-                                ripple,
-                                'background',
-                                this.tuiRipple || null,
-                            );
-                            this.renderer.appendChild(this.el, ripple);
+                take(1),
+                tap(() => {
+                    this.renderer.appendChild(element, ripple);
+                }),
+                switchMap(isTap => {
+                    const animation = ripple.animate(RIPPLE_ON_ANIMATION, {
+                        duration: tuiGetDuration(this.animationSpeed),
+                    });
+                    const animationEndOn$ = tuiTypedFromEvent(animation, 'finish');
+
+                    return isTap
+                        ? animationEndOn$
+                        : race(
+                              touchEnd$.pipe(switchMap(() => animationEndOn$)),
+                              animationEndOn$.pipe(switchMap(() => touchEnd$)),
+                          ).pipe(takeUntil(touchMove$), defaultIfEmpty(undefined));
+                }),
+                switchMap(() =>
+                    tuiTypedFromEvent(
+                        ripple.animate(RIPPLE_OFF_ANIMATION, {
+                            duration: tuiGetDuration(this.animationSpeed),
                         }),
-                        switchMap(isTap =>
-                            isTap
-                                ? animationEndOn$
-                                : race(
-                                      touchEnd$.pipe(switchMap(() => animationEndOn$)),
-                                      animationEndOn$.pipe(switchMap(() => touchEnd$)),
-                                  ),
-                        ),
-                        map(() => ripple),
-                    );
+                        'finish',
+                    ),
+                ),
+                tap(() => {
+                    element.removeChild(ripple);
```

Comment:
nitpicking: maybe write it in one line without `{}`? Since it's just one line there it would easier register as a single step in the pipeline at a glance.